### ISSUE TITLE
Mutation attempt in new Coroutines

### ISSUE THREAD

**Issue by mustafaozhan** [mockative/mockative#28]
I am trying to migrate to new coroutine memory model for my kotlin multiplatform project but the iOS target fails with following error:
```
kotlin.native.concurrent.InvalidMutabilityException: mutation attempt of frozen kotlin.collections.EmptySet@338bdd0
kotlin.native.concurrent.InvalidMutabilityException: mutation attempt of frozen kotlin.collections.EmptySet@338bdd0
	at kotlin.Throwable#<init>(/Users/teamcity1/teamcity_work/6326934d18cfe24e/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Throwable.kt:24)
	at kotlin.Exception#<init>(/Users/teamcity1/teamcity_work/6326934d18cfe24e/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Exceptions.kt:23)
	at kotlin.RuntimeException#<init>(/Users/teamcity1/teamcity_work/6326934d18cfe24e/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Exceptions.kt:34)
	at kotlin.native.concurrent.InvalidMutabilityException#<init>(/Users/teamcity1/teamcity_work/6326934d18cfe24e/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/native/concurrent/Freezing.kt:24)
	at <global>.ThrowInvalidMutabilityException(/Users/teamcity1/teamcity_work/6326934d18cfe24e/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/native/concurrent/Internal.kt:109)
	at <global>.Kotlin_AtomicReference_checkIfFrozen(Unknown Source)
	at kotlin.native.concurrent.AtomicReference#<init>(/Users/teamcity1/teamcity_work/6326934d18cfe24e/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/native/concurrent/Atomics.kt:237)
	at io.mockative.concurrency.AtomicReference#<init>(/Users/runner/work/mockative/mockative/mockative/src/nativeMain/kotlin/io/mockative/concurrency/AtomicReference.kt:7)
	at io.mockative.concurrency.AtomicRef#<init>(/Users/runner/work/mockative/mockative/mockative/src/commonMain/kotlin/io/mockative/concurrency/Atomic.kt:6)
	at io.mockative.concurrency.AtomicSet#<init>(/Users/runner/work/mockative/mockative/mockative/src/commonMain/kotlin/io/mockative/concurrency/AtomicSet.kt:8)
	at io.mockative.Mockable#<init>(/Users/runner/work/mockative/mockative/mockative/src/commonMain/kotlin/io/mockative/Mockable.kt:13)
	at com.oztechan.ccc.common.settings.SettingsRepositoryMock#<init>(/Users/mustafa.ozhan/Projects/Personal/CurrencyConverterCalculator/CCC/client/build/generated/ksp/ios/iosTest/kotlin/com/oztechan/ccc/common/settings/SettingsRepositoryMock.kt:19)
	at io.mockative#mock(/Users/mustafa.ozhan/Projects/Personal/CurrencyConverterCalculator/CCC/client/build/generated/ksp/ios/iosTest/kotlin/io/mockative/GeneratedMocks.kt:4)
	at com.oztechan.ccc.client.viewmodel.SettingsViewModelTest#<init>(/Users/mustafa.ozhan/Projects/Personal/CurrencyConverterCalculator/CCC/client/src/commonTest/kotlin/com/oztechan/ccc/client/viewmodel/SettingsViewModelTest.kt:45)
	at com.oztechan.ccc.client.viewmodel.$SettingsViewModelTest$test$0.createInstance#internal(/Users/mustafa.ozhan/Projects/Personal/CurrencyConverterCalculator/CCC/client/src/commonTest/kotlin/com/oztechan/ccc/client/viewmodel/SettingsViewModelTest.kt:41)
	at kotlin.native.internal.test.BaseClassSuite.TestCase#run(/Users/teamcity1/teamcity_work/6326934d18cfe24e/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/native/internal/test/TestSuite.kt:82)
	at kotlin.native.internal.test.TestRunner.run#internal(/Users/teamcity1/teamcity_work/6326934d18cfe24e/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/native/internal/test/TestRunner.kt:245)
	at kotlin.native.internal.test.TestRunner.runIteration#internal(/Users/teamcity1/teamcity_work/6326934d18cfe24e/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/native/internal/test/TestRunner.kt:271)
	at kotlin.native.internal.test.TestRunner#run(/Users/teamcity1/teamcity_work/6326934d18cfe24e/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/native/internal/test/TestRunner.kt:286)
	at kotlin.native.internal.test#testLauncherEntryPoint(/Users/teamcity1/teamcity_work/6326934d18cfe24e/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/native/internal/test/Launcher.kt:30)
	at kotlin.native.internal.test#main(/Users/teamcity1/teamcity_work/6326934d18cfe24e/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/native/internal/test/Launcher.kt:34)
	at <global>.Konan_start(/Users/teamcity1/teamcity_work/6326934d18cfe24e/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/native/internal/test/Launcher.kt:33)
	at <global>.Init_and_run_start(Unknown Source)
	at <global>.start_sim(Unknown Source)
	at <global>.0x0(Unknown Source)
```

Related PR: https://github.com/Oztechan/CCC/pull/567

**Comment by Nillerr** [comment]
Hi @mustafaozhan 

I have just released version `1.2.0` of Mockative (it might be 20-30 minutes until it's available on Maven Central), in which I tested it with the new memory manager without being able to replicate your issue. Would you mind checking whether the issue remains in this version?

**Comment by Nillerr** [comment]
Hi again

I discovered an issue in `1.2.0` causing the generated code to be invalid. I have resolved this issue in `1.2.1`, which is on its way out.

**Comment by mustafaozhan** [comment]
@Nillerr thanks a lot 🙂 I confirm that this issue is fixed with `1.2.1`
